### PR TITLE
precise step

### PR DESCRIPTION
**Pull request by tiemvanderdeure**
An attempt to avoid the worst floating point errors when calculating step.

@rafaqz let me know what you think about this heuristic

### CODE REVIEW

File: src/sources/commondatamodel.jl [hunk 1]
```diff
@@ -385,12 +385,16 @@ end
...
     length(index) == 1 && return Regular(zero(eltype(index))), Points()
-    
-    # Calculate step, avoiding as many floating point errors as possible
-    st = Float64(Base.step(Base.range(first(index), last(index); length = length(index))))
-    st_rd = round(st, digits = Base.floor(Int,-log10(eps(eltype(index))))) # round to nearest digit within machine epsilon
-    step = st_rd ≈ st ? st_rd : st # keep the rounded number if it is very close to the original
 
+    step = if eltype(index) <: AbstractFloat
+        # Calculate step, avoiding as many floating point errors as possible
+        st = Float64(Base.step(Base.range(first(index), last(index); length = length(index))))
```
Comment (rafaqz) [inline src/sources/commondatamodel.jl:391]:
```suggestion
        st = Base.step(Base.range(Float64(first(index)), Float64(last(index)); length = length(index)))
```

The `Float64` conversion needs to happen before the division in `range`, not after

**Comment by rafaqz** [comment]
I want to merge this but we probably should have a test of the edge case where the rounding actually does something, just to demonstrate that we do this at all

**Comment by rafaqz** [comment]
Something weird happening in the `resample` tests to cause a segmentation violation (signal 11).

https://github.com/rafaqz/Rasters.jl/actions/runs/9148702377/job/25152012187?pr=656#step:5:933

**Comment by tiemvanderdeure** [comment]
Funily enough I can't reproduce this by just running tests/resample.jl, so not sure what is going on?

**Comment by rafaqz** [comment]
Maybe its only segfaulting on a specific GDAL version. Welcome to the joy of C pointers

**Comment by rafaqz** [comment]
@tiemvanderdeure if you can rebase this and push again GDAL will probably work now
